### PR TITLE
chore: updated search to use createAmplifyGeocoder

### PR DESCRIPTION
**Pull request by thaddmt**
_Issue #, if available:_

_Description of changes:_
- Updated search docs to use `createAmplifyGeocoder`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by offlineprogrammer — APPROVED** [review]

File: docs/lib/geo/fragments/js/search.md [hunk 1]
```diff
@@ -25,10 +25,8 @@ The location search UI component (`maplibre-gl-geocoder`) requires a "geocoding
 Finally, add the location search UI component (`MaplibreGeocoder`) to the map.
 
 ```javascript
-import { AmplifyMapLibreRequest, AmplifyGeocoderAPI } from "maplibre-gl-js-amplify";
+import { AmplifyMapLibreRequest, AmplifyGeocoderAPI, createAmplifyGeocoder } from "maplibre-gl-js-amplify";
```
Comment (Amplifiyer) [inline docs/lib/geo/fragments/js/search.md:28]: With this change, we should remove the import `AmplifyGeocoderAPI` as well as line 23 in this file _The location search UI component (`maplibre-gl-geocoder`) requires a "geocoding API"..._ and update line 25 to use `createAmplifyGeocoder`

File: docs/lib/geo/fragments/js/search.md [hunk 2]
```diff
@@ -41,10 +39,7 @@ async function initializeMap() {
         zoom: 11,
     })
 
-    const geocoder = new MaplibreGeocoder(AmplifyGeocoderAPI, {
-        maplibregl: maplibregl,
-        showResultMarkers: true,
-    });
+    const geocoder = createAmplifyGeocoder());
     map.addControl(geocoder);
```
Comment (Amplifiyer) [inline docs/lib/geo/fragments/js/search.md:43]: we can probably combine these lines for brevity

File: docs/lib/geo/fragments/js/search.md [hunk 3]
```diff
@@ -60,9 +55,7 @@ You can also use [maplibre-gl-geocoder](https://github.com/maplibre/maplibre-gl-
 To do so, extract the html element using function `onAdd()` and attach it anywhere in your DOM instead of adding it via the map's `addControl()` function.
 
 ```javascript
-const geocoder = new MaplibreGeocoder(AmplifyGeocoderAPI, {
-    showResultMarkers: true,
-});
+const geocoder = new MaplibreGeocoder(AmplifyGeocoderAPI);
```
Comment (Amplifiyer) [inline docs/lib/geo/fragments/js/search.md:58]: Can we not directly use `createAmplifyGeocoder` here?

File: docs/lib/geo/fragments/js/search.md [hunk 4]
```diff
@@ -22,30 +22,22 @@ First, create a map onto which you want to add the location search UI component.
 
 The location search UI component (`maplibre-gl-geocoder`) requires a "geocoding API" to facilitate location-based search. `AmplifyGeocoderAPI` is geocoding API that is integrated with Amplify Geo. To define your own "geocoding API", see [AmplifyGeocoderAPI](https://github.com/aws-amplify/maplibre-gl-js-amplify/blob/main/src/AmplifyMapLibreGeocoder.ts).
```
Comment (Amplifiyer) [inline docs/lib/geo/fragments/js/search.md:23]: I still see this, do we need it?

Comment (thaddmt) [reply docs/lib/geo/fragments/js/search.md:23]: Added a line about how createAmplifyGeocoder creates a geocoder with pre defined settings

Comment (TreTuna) [reply docs/lib/geo/fragments/js/search.md:23]: I would probably be more on the side of just removing this, unless we are also adding a way for them to add a custom geocoder as well, though I didn't see that.

Comment (thaddmt) [reply docs/lib/geo/fragments/js/search.md:23]: Whoops actually I realized I misread your last comment, fixed

**Review by Milan-Shah — APPROVED** [review]

**Review by TreTuna — APPROVED** [review]

File: docs/lib/geo/fragments/js/search.md [hunk 5]
```diff
@@ -20,9 +20,7 @@ npm install -S @maplibre/maplibre-gl-geocoder maplibre-gl maplibre-gl-js-amplify
 
 First, create a map onto which you want to add the location search UI component. See the guide on [creating and displaying maps](~/lib/geo/maps.md).
 
-Then, add the location search UI component (`MaplibreGeocoder`) to the map by using `createAmplifyGeocoder()`.
-
-To define your own "geocoding API" for `MaplibreGeocoder`, see [AmplifyGeocoderAPI](https://github.com/aws-amplify/maplibre-gl-js-amplify/blob/main/src/AmplifyMapLibreGeocoder.ts).
+Then, using `createAmplifyGeocoder()` to get a new instance of `MaplibreGeocoder` and add the location search UI component to the map.
```
Comment (Amplifiyer) [inline docs/lib/geo/fragments/js/search.md:23]:
```suggestion
Then, use `createAmplifyGeocoder()` to get a new instance of `MaplibreGeocoder` and add the location search UI component to the map.
```

File: src/fragments/lib/geo/js/search.mdx [hunk 6]
```diff
@@ -8,7 +8,7 @@
 
 First, make sure you've provisioned a search index resource and configured your app using the instructions in either [Getting started](/lib/geo/getting-started) or [Use existing Amazon Location Service resources](/lib/geo/existing-resources) and you have already setup [displaying a map](/lib/geo/maps) in your application.
 
-To add a location search UI component to your map, you need to use the [maplibre-gl-geocoder](https://github.com/maplibre/maplibre-gl-geocoder) library with Amplify Geo's `maplibre-gl-js-amplify` package. `maplibre-gl-js-amplify` makes it easy to integrate MapLibre with Amplify Geo.
+To add a location search UI component to your map, you need to use the [maplibre-gl-geocoder](https://github.com/maplibre/maplibre-gl-geocoder) library with Amplify Geo's `maplibre-gl-js-amplify` package. `maplibre-gl-js-amplify` makes it easy to integrate MapLibre with Amplify Geo by exporting `createAmplifyGeocoder()` a utility function that return an instance of `maplibre-gl-geocoder` with some pre-defined settings.
```
Comment (Amplifiyer) [inline src/fragments/lib/geo/js/search.mdx:11]:
```suggestion
To add a location search UI component to your map, you can use the [maplibre-gl-geocoder](https://github.com/maplibre/maplibre-gl-geocoder) library. `maplibre-gl-js-amplify` package makes it easy to integrate `maplibre-gl-geocoder` with Amplify Geo by exporting a utility function `createAmplifyGeocoder()` that returns an instance of `maplibre-gl-geocoder` with some pre-defined settings and supports all the [options](https://github.com/maplibre/maplibre-gl-geocoder/blob/main/API.md#parameters) for customizing the UI component
```

**Review by Amplifiyer — APPROVED** [review]

**Pull request merged**